index.js

```diff
@@ -73,16 +75,42 @@ function getMainArgs() {
 }
 
 const protoKey = '__proto__';
-function storeOptionValue(options, longOption, value, result) {
-  const optionConfig = options[longOption] || {};
+
+function storeOption({
+  strict,
+  options,
+  result,
+  longOption,
+  shortOption,
+  optionValue,
+}) {
+  const hasOptionConfig = ObjectHasOwn(options, longOption);
+
+  const optionConfig = hasOptionConfig ? options[longOption] : {};
+
+  if (strict) {
+    if (!hasOptionConfig) {
+      throw new ERR_PARSE_ARGS_UNKNOWN_OPTION(shortOption == null ? `--${longOption}` : `-${shortOption}`);
```

Comment:
Me too! But we had some _long_ conversions about this, and the reasoning for doing weak test against null to avoid contamination of undefined.

- https://github.com/pkgjs/parseargs/pull/74#discussion_r818969649

Reply:
Also: https://github.com/pkgjs/parseargs/pull/74#discussion_r819093456

Reply:
The point is to check for both; that’s why the comparison is loose.

Reply:
works for me.